core/src/main/scala/org/apache/spark/SparkHadoopWriter.scala

```diff
@@ -106,18 +107,25 @@ class SparkHadoopWriter(@transient jobConf: JobConf)
     val taCtxt = getTaskContext()
     val cmtr = getOutputCommitter()
     if (cmtr.needsTaskCommit(taCtxt)) {
-      try {
-        cmtr.commitTask(taCtxt)
-        logInfo (taID + ": Committed")
-      } catch {
-        case e: IOException => {
-          logError("Error committing the output of task: " + taID.value, e)
-          cmtr.abortTask(taCtxt)
-          throw e
+      val outputCommitCoordinator = SparkEnv.get.outputCommitCoordinator
+      val conf = SparkEnv.get.conf
+      val canCommit: Boolean = outputCommitCoordinator.canCommit(jobID, splitID, attemptID)
+      if (canCommit) {
```

Comment:
> Otherwise, we probably need some new state (e.g. "task gave up") that doesn't count as an error.

We have the `TaskEndReason` mechanism for special-handling of certain kinds of task failures, so maybe we can add a new reason and update the corresponding [handlers in TaskSetManager](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala#L605) and [DAGScheduler](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala#L941).

We might still end up having to throw an exception, but we can throw a more specific one that we can catch and transform into the right TaskEndReason.